### DISCUSSION TITLE
Any good examples of the JFreechart extension?

### DISCUSSION THREAD

**Discussion by wigoe** [optimatika/ojAlgo#488] [General, ojAlgo and related code]
I use JFreechart quite a bit and wanted to see how ojalog incorporates it into its algorithms/code.  Any examples?

**Comment by apete** [comment]
ojAlgo in no way incorporates jfreechart.

The ojAlgo-jfreechart extension artifact contains some standardsed chart builders/factories. Some of which make use of ojAlgo data types (like time series).

The chart in the lower part of this screenshot

![](https://www.optimatika.se/wp-content/uploads/2022/08/Screenshot-2022-08-24-at-11.13.12.png)

is produced using high level code like this:
```java
    public static JFreeChartAdaptor makeHistoryAndForecast(final BLMarket market) {

        HistoryAndForecastBuilder retVal = new HistoryAndForecastBuilder();

        for (BLAssetClassCategory instrument : market.getAssetClassCategories()) {
            CalendarDateSeries<RandomNumber> series = instrument.getSeriesWithForecast();
            retVal.add(series);
        }

        retVal.legend(false);

        return retVal.build();
    }
```

**Reply by wigoe** [reply]
OK thanks very nice!
